### PR TITLE
fix compilation without deprecated APIs

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by alfredh** [comment]
can you please include which OS and openssl version the warning comes from ?

please include the relevant build output.

**Comment by neheb** [comment]
Output is basically the opposite of this: https://travis-ci.org/github/baresip/baresip/jobs/715650996

Tested on OpenWrt with OpenSSL deprecated APIs disabled.

This can be simulated by passing -DOPENSSL_API_COMPAT=0x10100000L and -Werror=implicit-function-declaration .

**Comment by alfredh** [comment]
can you make sure that the patch works with all supported versions of OpenSSL:

- OpenSSL version 1.0.2
- OpenSSL version 1.1.0
- OpenSSL version 1.1.1
- LibreSSL version 2.x

https://github.com/baresip/baresip/wiki/Supported-platforms#supported-versions-of-openssl

File: modules/debug_cmd/debug_cmd.c [hunk 1]
```diff
@@ -56,7 +60,7 @@ static int print_system_info(struct re_printf *pf, void *arg)
 
 #ifdef USE_OPENSSL
 	err |= re_hprintf(pf, " OpenSSL:  %s\n",
-			  SSLeay_version(SSLEAY_VERSION));
+			  OpenSSL_version(OPENSSL_VERSION));
```
Comment (alfredh) [inline modules/debug_cmd/debug_cmd.c:63]: is it not better to check for the version that the function was added ?

```c
#if VERSION >= X
OpenSSL_version(OPENSSL_VERSION);
#else
SSLeay_version(SSLEAY_VERSION);
#endif
```

**Comment by alfredh** [comment]
this fails to build on OpenBSD 6.6. with LibreSSL 3.0.2:

```
modules/debug_cmd/debug_cmd.c:11:1: warning: "OPENSSL_VERSION" redefined
In file included from modules/debug_cmd/debug_cmd.c:9:
/usr/include/openssl/crypto.h:329:1: warning: this is the location of the previous definition
```

File: modules/debug_cmd/debug_cmd.c (hunk 1 above)
Comment (neheb) [reply modules/debug_cmd/debug_cmd.c:63]: That's done above.

**Comment by neheb** [comment]
> 
> 
> this fails to build on OpenBSD 6.6. with LibreSSL 3.0.2:
> 
> ```
> modules/debug_cmd/debug_cmd.c:11:1: warning: "OPENSSL_VERSION" redefined
> In file included from modules/debug_cmd/debug_cmd.c:9:
> /usr/include/openssl/crypto.h:329:1: warning: this is the location of the previous definition
> ```

I'll fix when I can.

**Comment by alfredh** [comment]
hi @neheb when are you planning to update the PR ?

**Comment by neheb** [comment]
Maybe now?

**Pull request closed without merging**